pkg/crossplane/service_mariadb_database.go

```diff
@@ -96,7 +96,13 @@ func (msb MariadbDatabaseServiceBinder) Bind(ctx context.Context, bindingID stri
 		return nil, err
 	}
 
-	creds := createCredentials(endpoint, bindingID, pw, msb.instance.ID())
+	mp := ""
+	mbyte, ok := secret.Data[MetricsPortKey]
+	if ok {
+		mp = string(mbyte)
+	}
```

Comment:
Would it maybe make sense to include this conversion in the createCredentials function?

Reply:
I personally wouldn't, as this would make `createCredentials` inconsistent. (Why should only the metrics port be a byte slice). But I realized that the `ok` checking became useless, so that way it should look a little cleaner.